### ISSUE TITLE
Complete Oracle tests

### ISSUE THREAD

**Issue by gintsgints** [typeorm/typeorm#452]
I would like to create this thread for issues on Oracle test execution.

**Comment by pleerock** [comment]
we are lucky if we can setup docker container with oracle inside and run it on CI. btw are you using oracle?

**Comment by gintsgints** [comment]
Some questions I have:
- Why normal tests includes sample2-one-to-one.ts (line 24) file with only MySQL driver? This means you cannot skip MySQL tests?
- Probably there will be lot of issues about ORA-00972: identifier is too long. Need to find out to change name generator for it

**Comment by gintsgints** [comment]
"setup docker container with oracle inside and run it on CI" is big aim :)

Oracle answered that it is ok to use they image at store. But this requests signing up to docker store. Not sure how to do this with travis.

https://github.com/oracle/docker-images/issues/367#issuecomment-298252450

**Comment by pleerock** [comment]
1. `sample*` tests were first tests I developed and at that time only mysql were supported. We need to refactor those tests to make them to work with all drivers.

2. `ORA-00972: identifier is too long.` What is this issue about? Identifier of what? table? column? alias,?

I remember I setup oracle in docker on my mac machine. But it was a long time ago. I think we can setup docker for oracle too (but images are huuuuge).

**Comment by gintsgints** [comment]
This time it was column alias.

**Comment by pleerock** [comment]
this happens sometimes on a really very very long generated names, probably only on many-to-many tables. You can setup custom column names for many to many tables.

**Comment by gintsgints** [comment]
I already using custom generator for project I use typeorm. Option could be setup custom generator for tests too.

**Comment by gintsgints** [comment]
Dealt with column names and Join table names so far. Now comes index:

```
query failed: CREATE  INDEX "ind_e25153a614c8d27da45a08a506f" ON "ps_cats"("sample4PostCategoryId")
      1) "before all" hook: reloadDatabase
```

gives

```
Error: ORA-00972: identifier is too long
```

**Comment by gintsgints** [comment]
@pleerock Please check my pull requests.

**Comment by pleerock** [comment]
@gintsgints done. Can you please setup docker container for travis so we can run oracle tests?

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.23`. You can try it via `npm i typeorm@next`.

**Comment by Kononnable** [comment]
I'm not sure what was fixed here. According to ormconfig.travis.json Travis is skipping testing on oracle.

I've managed to get oracle db working on travis. However there are few drawbacks. Main one is that oracle tests can only work on builds made by collabolators(travis encrypted variables), so if something is broken by PR in oracle driver it would be known only after merging.

I can make a PR for it if you're still interested in testing oracle on travis.
